rfcs/proposed/passed_directly_API/README.md

```diff
@@ -0,0 +1,144 @@
+at first implementation.
+
+### Implementation details
+To make this robust, we will follow an C++17 updated version of what is discussed in
+[Eric Niebler's Post](https://ericniebler.com/2014/10/21/customization-point-design-in-c11-and-beyond/), using a
+callable, and using an `inline constexpr` to avoid issues with ODR and to avoid issues with resolving customization
+points when not separating the call to two steps with a `using` statement first.
+
+## Alternatives considered
+### Public trait struct explicit specialization
+We could simply make public our internal structure `oneapi::dpl::__ranges::is_passed_directly` as
+`oneapi::dpl::is_passed_directly` for users to specialize to define rules for their types. This would be a similar
+mechanism to `sycl::is_device_copyable`. The implementation details of this option should avoid some complexity required
+to properly implement the customization point.
+
+However, as we have learned from experience within oneDPL, explicit specialization of a structure in another library's
+namespace makes for maintenance problems. It either requires lots of closing of nested namespaces, opening of the
+external library's namespace for the specialization or it requires separating these specializations to a separate
+location removed from the types they are specializing for. OneDPL has chosen to use the later, which can be seen in
+`include/oneapi/dpl/pstl/hetero/dpcpp/sycl_traits.h`. This has made for several errors where changes to structures
+should have included changes to sycl_traits, but did not, and needed to be fixed later.
+
+In an effort to avoid this same issue for our users, we propose a similar method but instead with a constexpr
+customization point, allowing the user to override that customization point within their own namespace as a free
+function.
+
+### Require specifically named typedef / using in user's type
+We could simply make official our requirements for user's types to include a typedef or using statement to define if the
+type is passed directly like `using is_passed_directly = std::true_type;`, where the absence of this would be equivalent
+to a `std::false_type`. 
+
+However, this clutters the user type definitions with specifics of oneDPL. It also may not be as clear what this
+signifies for maintenance of user code without appropriate comments describing the details of oneDPL and SYCL. Users
+have expressed that this is undesirable.
+
+### Testing
+We will need a detailed test checking both positive and negative responses to `is_passed_directly_to_sycl_kernels` come
+as expected, with custom types and combinations of iterators, usm pointers etc.
+
+## Open Questions
+
+Is there a better / more concise name than `is_passed_directly_to_sycl_kernels` we can use which properly conveys the
+meaning to the users?
+
+Should we be targeting Experimental or fully supported with this proposal?
+ (Do we think user feedback is required to solidify an interface / experience?)
```

Comment:
I guess the general design of passed directly has been tested internally pretty well at least.

Reply:
Agreed.  From my perspective the only reason to keep it in experimental to start with is if we are uncertain of the exact API specifics or to find any unexpected gotchas with the approach of using a customization point generally as opposed to some other option.  

Lets see what others have to say, but I'm leaning toward targeting supported, and just adding it to the specification directly.

Reply:
I agree that there is not much need for an experimental phase. But a POC with practical usage outside of oneDPL is necessary I think.